govcd/vapp_test.go

```diff
@@ -1557,3 +1558,101 @@ func (vcd *TestVCD) Test_AddNewVMWithComputeCapacity(check *C) {
 	_, err = adminVdc.SetAssignedComputePolicies(types.VdcComputePolicyReferences{VdcComputePolicyReference: beforeTestPolicyReferences})
 	check.Assert(err, IsNil)
 }
+
+func (vcd *TestVCD) testUpdateVapp(check *C, vapp *VApp, name, description string, vms []string) {
+
+	var err error
+
+	// Make sure that at least one value was set
+	bothEmpty := description == "" && name == ""
+	check.Assert(bothEmpty, Equals, false)
+
+	if name == "" && description != "" {
+		printVerbose("testing vapp.UpdateDescription(%s)\n", description)
+		err = vapp.UpdateDescription(description)
+		check.Assert(err, IsNil)
+	}
+	if description == "" && name != "" {
+		printVerbose("testing vapp.Rename(%s)\n", name)
+		err = vapp.Rename(name)
+		check.Assert(err, IsNil)
+	}
+	if description != "" && name != "" {
+		printVerbose("testing vapp.UpdateNameDescription(%s, %s)\n", name, description)
+		err = vapp.UpdateNameDescription(name, description)
+		check.Assert(err, IsNil)
+	}
+
+	if name == "" {
+		name = vapp.VApp.Name
+	}
+	if description == "" {
+		description = vapp.VApp.Description
+	}
+
+	// Get a fresh copy of the vApp
+	vapp, err = vcd.vdc.GetVAppByName(name, true)
+	check.Assert(err, IsNil)
+
+	check.Assert(vapp.VApp.Name, Equals, name)
+	check.Assert(vapp.VApp.Description, Equals, description)
+	// check that the VMs still exist after vApp update
+	for _, vm := range vms {
+		printVerbose("checking VM %s\n", vm)
+		_, err = vapp.GetVMByName(vm, true)
+		check.Assert(err, IsNil)
+	}
+}
+
+func (vcd *TestVCD) Test_UpdateVappNameDescription(check *C) {
+
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	vappName := check.TestName()
+	vappDescription := vappName
+	newVappName := vappName + "_new"
+
+	newVappDescription := vappName + " desc"
+	// Compose VApp (the description, by default, is the vapp name)
+	vapp, err := makeEmptyVapp(vcd.vdc, vappName)
+	check.Assert(err, IsNil)
+	AddToCleanupList(vappName, "vapp", "", "Test_RenameVapp")
+
+	check.Assert(vapp.VApp.Name, Equals, vappName)
+	check.Assert(vapp.VApp.Description, Equals, vappName)
+
+	// Need a slight delay for the vApp to get the links that are needed for renaming
+	time.Sleep(time.Second)
+
+	// change description
+	vcd.testUpdateVapp(check, vapp, "", newVappDescription, nil)
```

Comment:
I may have missed it, but we need a test which tries to re-set description to empty.

Reply:
added